hazelcast/include/hazelcast/client/ClientConfig.h

```diff
@@ -80,12 +88,14 @@ namespace hazelcast {
             ClientConfig &addAddresses(const std::vector<Address> &addresses);
 
             /**
+            * @deprecated Please use {@link ClientNetworkConfig#getAddresses()}.
+            *
             * Returns set of the initial addresses.
             * Client will use this vector to find a running Member, connect to it.
             *
             * @return vector of addresses
             */
-            std::set<Address, addressComparator> &getAddresses();
+            std::set<Address> getAddresses();
```

Comment:
Isn't this a breaking change ? Return type is different.

Reply:
I typedef less<Address> operator to addressComparator and added ClientConfigTest::testGetAddresses to make sure that it compiles and works and no compatibility issue exist..